Monika After Story/game/script-greetings.rpy

```diff
@@ -3068,17 +3071,47 @@ label greeting_back_from_work:
         menu:
             m "How was work today?{fast}"
 
-            "Amazing.":
+            "Amazing!":
+                if not persistent._mas_pm_got_promotion:
+                    $ promoted_recently = False
+                else:
+                    $ promoted_recently = datetime.date.today() < persistent._mas_pm_got_promotion + datetime.timedelta(days=180)
                 m 1sub "That's {i}amazing{/i}, [player]!"
                 m 1hub "I'm really happy that you had such a great day!"
+
+                m 1sua "What made it such an amazing day?{nw}"
+                menu:
+                    m "What made it such an amazing day?{fast}"
+
+                    "I moved up!":
+                        if promoted_recently:
+                            m 3suo "Wow! Again?!"
+                            m 3sub "You just got a promotion not that long ago...{w=0.5}you must really be doing amazing work!"
+                            m 1huu "I'm so, {w=0.3}so proud of you, [mas_get_player_nickname()]~"
+                        else:
+                            $ player_nick = mas_get_player_nickname()
+                            m 3suo "Wow! Congratulations [player_nick], {w=0.1}{nw}"
+                            extend 3hub "I'm so proud of you!"
+                            m 1euu "I knew you could do it~"
+                        $ persistent._mas_pm_got_promotion = datetime.date.today()
+
+                    "I got a lot done!":
+                        m 3hub "That's great, [mas_get_player_nickname()]!"
+
+                    "It was just an amazing day.":
+                        m 3hub "That's great to hear!"
+
                 m 3eua "I can only imagine how well you must work on days like that."
-                m 1hua "...Maybe you'll even move up a bit soon!"
+                if not promoted_recently:
+                    m 1hua "...Maybe you'll even move up a bit soon!"
```

Comment:
She'll say this line even if you just got your promotion. We probably need another var to check it, and either not show this line at all, or adjust it accordingly, like `"...Maybe you'll even move up a bit again soon!"`